File: tests/dynamic_fixtures/php/xss_unsupported.php
<?php
// XSS — unsupported fixture.
// Entry is a class method; test sets confidence = Low.
// Expected verdict: Unsupported

class TemplateEngine {
    public function render($input) {
        echo '<html><body>' . $input . '</body></html>' . "\n";
    }
}

File: tests/dynamic_fixtures/rust/xss_unsupported.rs
/// XSS — unsupported entry-kind fixture.
///
/// The vulnerable logic lives inside a struct method. The test creates a Diag
/// with Confidence::Low, so `HarnessSpec::from_finding` returns
/// `Err(UnsupportedReason::ConfidenceTooLow)`.
///
/// Expected verdict: Unsupported(ConfidenceTooLow)
/// Cap: HTML_ESCAPE
pub struct PageRenderer;

impl PageRenderer {
    pub fn render(&self, user_input: &str) -> String {
        // Vulnerable: no HTML escaping.
        format!("<div>{}</div>", user_input)
    }
}
